test/python/algorithms/test_grover.py

```diff
@@ -251,19 +266,90 @@ def test_max_probability(self):
         oracle = QuantumCircuit(2)
         oracle.cz(0, 1)
         problem = AmplificationProblem(oracle, is_good_state=["11"])
-        grover = Grover(quantum_instance=self.qasm)
+        with self.assertWarns(PendingDeprecationWarning):
+            grover = Grover(quantum_instance=self.qasm)
         result = grover.amplify(problem)
         self.assertEqual(result.max_probability, 1.0)
 
     def test_oracle_evaluation(self):
         """Test oracle_evaluation for PhaseOracle"""
         oracle = PhaseOracle("x1 & x2 & (not x3)")
         problem = AmplificationProblem(oracle, is_good_state=oracle.evaluate_bitstring)
-        grover = Grover(quantum_instance=self.qasm)
+        with self.assertWarns(PendingDeprecationWarning):
+            grover = Grover(quantum_instance=self.qasm)
+        result = grover.amplify(problem)
+        self.assertTrue(result.oracle_evaluation)
+        self.assertEqual("011", result.top_measurement)
+
+    def test_circuit_result_sampler(self):
```

Comment:
To ensure the sampler works like the quantum instance before, would it be better to structure the tests like
```python
@data(True, False)
def test_something(self, use_sampler):
    """Test a thing here."""
    # as before ...

    if use_sampler:
        grover = Grover(..., sampler=self._sampler)
    else:
        with self.assertWarns(PendingDeprecationWarning):
            grover = Grover(..., quantum_instance=self._quantum_instance)
            
    # ... as before
```
This way we would test all the same things as before, with the same tests that existed before 🙂